File: Project-Euler/Stats/euler5/euler5.jl
#=
2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
=#
#Answer = 232_792_560

#Terminal: julia euler5.jl 20 5000000000
div = parse(Int, ARGS[1])
limit = parse(Int, ARGS[2])

function divisible(div, limit)
  for x = div + 1 : limit #start from above div
    evenlyDivis = true
    for i = 2 : div
      n = x % i
      if !(n == 0)
        evenlyDivis = false
        break
      end
    end
    if evenlyDivis
      return x
    end
  end
end

println(divisible(div, limit))

File: Project-Euler/Stats/euler5/euler5.py
import sys

def divisible(div, limit):
	evenlyDivis = True
	n = 0
	for x in xrange(div+1, limit):
		evenlyDivis = True
		for i in xrange(2, div):
			n = x % i
			if not(n == 0):
				evenlyDivis = False
				break
		if evenlyDivis:
			return x

div = int(sys.argv[1])
limit = int(sys.argv[2])
print divisible(div, limit)
